core/cap-0046-01.md

```diff
@@ -297,26 +362,26 @@ Relative to requirements listed in this CAP, WASM addresses many of them:
     - Learnability: WASM is not as familiar as EVM but is relatively widely known and appears easy to learn.
 
-### Rationale for value / object split
+### Rationale for host value / host object split
 
-The split between values (which can traverse the host/guest interface) and objects (which remain on the host side and are managed by host functions) is justified as a response to a number of observations we made when considering existing blockchains:
+The split between host value types (`Val`s that can traverse the host/guest interface) and host objects (that remain on the host side, are identified only by handles, and are managed by host functions) is justified as a response to a number of observations we made when considering existing blockchains:
 
   - Many systems spend a lot of guest code footprint (time and space) implementing data serialization and deserialization to and from opaque byte arrays. This code suffers from a variety of problems:
-    - It is often to and from an opaque format, making a contract's data difficult to browse or debug, and making SDKs that invoke contracts need to carry special code to serialize and deserialize data for the contract.
+    - It is often to and from an opaque, non-standard or contract-specific format, making a contract's data difficult to browse or debug, and making SDKs that invoke contracts need to carry special code to serialize and deserialize data for the contract.
     - It is often coupled to a specific version or layout of a data structure, such that data cannot be easily be migrated between versions of a contract.
     - It requires that a contract potentially contains extra copies of serialization support code for the formats used by any contracts it calls.
     - It is often intermixed with argument processing and contract logic, representing a significant class of security problems in contracts. 
     - It is usually unshared code: each contract implements its own copy of serialization and deserialization, and does so inefficiently in the guest rather than efficiently on the host. 
 
   - Similarly, when guest code is CPU-intensive it is often performing numerical or cryptographic operations which would be better supported by a common library of efficient (native) host functions.
 
-  - As of this writing, WASM defines no mechanism of directly sharing code, which makes it impossible to reuse common guest functions needed by many contracts. Sharing common host functions is comparatively straightforward, and much more so if we define a common data model on which host functions operate.
+  - As of this writing, WASM defines no standardized mechanism of directly sharing code, which makes it impossible to reuse common guest functions needed by many contracts. Sharing common host functions is comparatively straightforward, and much more so if we define a common data model on which host functions operate.
```

Comment:
> WASM defines no standardized mechanism of directly sharing code

I'm not expert of various wasm standards and proposals, but is this what Wasm component model is designed for? If so, maybe it's better to say there is no mature mechanism for standardized code/module sharing.

Reply:
It's immature, but sure, made a note.